spec.md

```diff
@@ -840,6 +841,25 @@ If the image had already been deleted or did not exist, a `404 Not Found` respon
 
 > for more details, see: [compatibility.md](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md#backward-compatibility)
 
+### Extensions
+
+Extensions to the distribution-spec's registry API are for the purpose of defining OPTIONAL endpoint behaviour.
+Extensions are only complementary to the distribution-spec only having additive features.
+Client can discover the extensions available, if any.
+
+These extensions are similar concept to the use of "feature flags".
+
+See the [`./ext/` directory](./ext/) for extensions available and their definitions.
```

Comment:
@vbatts Do you have any thoughts on having an API-driven discovery mechanism for back-compatibility between versions of extensions, or do you fully intend all new versions of extensions to be completely independent?

Reply:
is this to say, clients could stay using the older version of an extension?
If so, no. That is up to the registry maintainer. I think the `$name` of any given extension ought to link to the its markdown doc describing it. If it changes, then the `$name` and doc should change. The registry owner can choose the lifecycle they support here.

Reply:
Hmm... I have concerns we'll run into the same problems that we're running into with `/v2` in regards to back and forward compatibility. At minimum, I think we should have extensions have a defined way of publishing support for specific things or versions

Reply:
would `/v2/_ext/oci-artifacts/v1/{namespace}/repo/manifests/{digest}/links?n=10` be inline with what we're discussing here?
- oci-artifacts would have a links extension.
- it would have a version `v1`
- it would follow the `namespace/repo` path consistency, but be rooted in a named extension `/oci-artifacts`